src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -319,15 +387,58 @@ private boolean post(final JSONObject payload, final String type) {
         return status;
     }
 
-    @Override
-    public boolean validate() throws IOException, ServletException {
-        String urlParameters = "?api_key=" + apiKey;
+    /**
+     * Posts a given {@link JSONObject} payload to the Datadog API, using the
+     * user configured apiKey.
+     *
+     * @param payload - A JSONObject containing a specific subset of a builds metadata.
+     * @return a boolean to signify the success or failure of the HTTP POST request.
+     */
+    public boolean sendLogs(String payload) {
+        if(this.isLogIntakeConnectionBroken()){
+            logger.severe("Your client is not initialized properly");
+            return false;
+        }
+
+        if(this.getLogIntakeUrl() == null || this.getLogIntakeUrl().isEmpty()){
+            logger.severe("Datadog Log Intake URL is not set properly");
+            throw new RuntimeException("Datadog Log Collection Port not set properly");
+        }
+
+        return postLogs(this.getLogIntakeUrl(), getApiKey(), payload);
+    }
+
+    @SuppressFBWarnings("REC_CATCH_EXCEPTION")
+    private boolean postLogs(String url, Secret apiKey, String payload) {
+        if(payload == null){
+            logger.fine("No payload to post");
+            return true;
+        }
+
         HttpURLConnection conn = null;
-        boolean status = true;
         try {
-            // Make request
-            conn = getHttpURLConnection(new URL(url + VALIDATE + urlParameters));
-            conn.setRequestMethod("GET");
+            URL logsEndpointURL = new URL(url);
+            logger.fine("Setting up HttpURLConnection...");
+            conn = getHttpURLConnection(logsEndpointURL);
+            conn.setRequestMethod("POST");
+            conn.setRequestProperty("Content-Type", "application/json");
+            conn.setRequestProperty("DD-API-KEY", Secret.toString(apiKey));
+            conn.setRequestProperty("User-Agent", String.format("Datadog/%s/jenkins Java/%s Jenkins/%s",
+                    getDatadogPluginVersion(),
+                    getJavaRuntimeVersion(),
+                    getJenkinsVersion()));
+            DatadogUtilities.severe(logger, null, String.format("Datadog/%s/jenkins Java/%s Jenkins/%s",
+                    getDatadogPluginVersion(),
+                    getJavaRuntimeVersion(),
+                    getJenkinsVersion()));
```

Comment:
Is this supposed to be here?

Reply:
Definitely not and still on master, I'll remove it now 👍